shuup/admin/views/select.py

```diff
@@ -72,21 +74,35 @@ def get_data(self, request, *args, **kwargs):
         model_name = request.GET.get("model")
         if not model_name:
             return []
+
         cls = apps.get_model(model_name)
         qs = cls.objects.all()
         shop = self.request.shop
         qs = self._filter_query(cls, qs, shop)
         self.init_search_fields(cls)
         if not self.search_fields:
             return [{"id": None, "name": _("Couldn't get selections for %s.") % model_name}]
+
         if request.GET.get("search"):
             query = Q()
             keyword = request.GET.get("search", "").strip()
             for field in self.search_fields:
                 query |= Q(**{"%s__icontains" % field: keyword})
+
             if issubclass(cls, Contact) or issubclass(cls, get_user_model()):
                 query &= Q(is_active=True)
-            qs = qs.filter(query).distinct()
+
+            qs = qs.filter(query)
+
+        search_mode = request.GET.get("searchMode")
+        if search_mode and search_mode == "main" and issubclass(cls, Product):
+            qs = qs.filter(mode__in=[
+                ProductMode.SIMPLE_VARIATION_PARENT,
+                ProductMode.VARIABLE_VARIATION_PARENT,
+                ProductMode.NORMAL
+            ])
```

Comment:
Could this be `ProductQuerySet::top_level_products()` or similar, so this could be changed to `qs = qs.top_level_products()`? it would make it more re-usable.

Reply:
I guess so.